### PR TITLE
fix: sending query params on redirection from signup to login page

### LINKED ISSUE #11932: [Bug]: Restricted signup error should be displayed when uninvited user tries to login/signup
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When an uninvited user tries to log in with Google or GitHub the error message is not displayed on UI It just Displayed in Fat container logs

### Steps To Reproduce

1. In Localhost 
2. Try to login with the google login
3. Make sure you are an uninvited user
4. the user will be navigated back to the login page without any error message which would be confusing to users

### Public Sample App

_No response_

### Version

local

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Redirection from signup to login page doesn't include query params. This redirection happens when form login is disabled. This has been fixed in this PR.

Fixes #11932 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Checked if the redirection passes on the query parameters from signup page to login page.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: hotfix/signup-redirection 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 70.59 **(0)** | 100 **(0)** | 92.38 **(-0.95)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/src/ce/pages/UserAuth/SignUp.tsx [hunk 1]
```diff
@@ -88,7 +88,12 @@ export function SignUp(props: SignUpFormProps) {
   const history = useHistory();
   useEffect(() => {
     if (disableLoginForm) {
-      history.replace(AUTH_LOGIN_URL);
+      const currentURL = new URL(window.location.href);
+      const searchParams = currentURL.searchParams;
+      history.replace({
+        pathname: AUTH_LOGIN_URL,
+        search: searchParams.toString(),
+      });
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/pages/UserAuth/SignUp.tsx:96]:
```
const search = new URL(window.location.href)?.searchParams?.toString();
history.replace({
        pathname: AUTH_LOGIN_URL,
        search,
});
```

Does this makes sense?

**Review by IAmAnubhavSaini — CHANGES_REQUESTED** [review]

File: app/client/src/ce/pages/UserAuth/SignUp.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/ce/pages/UserAuth/SignUp.tsx]: done

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**